### PR TITLE
Chore: update dependencies

### PR DESCRIPTION
**Pull request by kevinwcyu**
Changes:
- Updates node version in `.nvmrc` to current LTS version
- Added `moduleNameMapper` for css files after updating jest dependencies
- Updated rollup to v4
- Most changes in the `src/*` files are fixing deprecations

**Note to reviewer**:
- View with "Hide whitespace" enabled

### CODE REVIEW

File: src/tests/setupTests.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 // enables assertions such as toBeInTheDocument to be used in our tests
 import '@testing-library/jest-dom';
+import { TextEncoder, TextDecoder } from 'util';
+
+Object.assign(global, { TextDecoder, TextEncoder });
```
Comment (iwysiu) [inline src/tests/setupTests.ts:5]: Out of curiosity (not a nit) what's this doing?

**Review by iwysiu — APPROVED** [review]
😮 so many updates

**Comment by iwysiu** [comment]
Also, your call, but if you're planning to cut a release for these changes you could put the version/changelog update in this pr

File: src/tests/setupTests.ts (hunk 1 above)
Comment (kevinwcyu) [reply src/tests/setupTests.ts:5]: This was something that used to be available with `jest-dom`, but after updating it needs to be added manually.

**Pull request merged**